src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -149,6 +149,7 @@ public void preloadPreparedStatements()
                     clientState.setKeyspace(keyspace);
 
                 Prepared prepared = parseAndPrepare(query, clientState, false);
+                prepared.cacheEntrySize = measureCacheEntrySize(id, prepared);
```

Comment:
@bereng what about putting `measureCacheEntrySize` into `prepareAndParse`? That method would be renamed to `measureSize` or similar. Are there any cases when we want to prepare and parse and _not_ have it measured? I think that by the act of preparing and parsing, it should be all measured already so we do not need to deal with it anymore.

Reply:
Yes I share your concern, I thought about that as well. But I decided I didn't want to burden pstmnts with the extra sizing in cases where it might not be needed. I couldn't come up with any concrete examples but any customer apps, side projects, other usages of code, some pstmnts generator stress tool, we don't know what happens in C* forks,  etc would be some. Wdyt?

Reply:
perf tests need to compute it anyway, no? Otherwise I think that measuring the size of it is pretty innocent operation.

Reply:
I don't have a strong opinion here so :shrug: 

But when moving `measureCacheEntrySize(Object key, Prepared value)` I noticed it measures both the key and the value. Having a pstmnt calculate and hold it's size sgtm. But the pstmnt having it's size calculated with the size of a random key factored in makes no sense. Specially when that key is going to be implementation specific to a particular cache. That's not right.

What about having pstmnt having it's size precalculated and on `QueryProcessor` we pull that and calculate the cache entry size?

Reply:
javadoc on weigher() says:

_Specifies the weigher to use in determining the weight of entries. Entry weight is taken into consideration by maximumWeight(long) when determining which entries to evict, and use of this method requires a corresponding call to maximumWeight(long) prior to calling build. Weights are measured and recorded when entries are inserted into or updated in the cache, and are thus effectively static during the lifetime of a cache entry._

That is all true, but they think that key will / might be of different size. We have keys of a constant size because it is md5sum so we just dont need to take it into account, no?

Reply:
We spoke on Slack and it seems we agree having pstmnts hold the stmnt size alone is best, adding the key size later when needed.